cmd/cloudspaces.go

```diff
@@ -789,21 +786,94 @@ func validateBidPrice(bidPrice string) (string, error) {
 	return formatted, nil
 }
 
-// parseNodepoolParams parses nodepool parameters in format key1=value1,key2=value2
+// spotNodePoolParams represents the structure for spot nodepool parameters
+type spotNodePoolParams struct {
+	Desired     interface{} `json:"desired"`
+	ServerClass string      `json:"serverclass"`
+	BidPrice    float64     `json:"bidprice"`
+}
+
+// parseNodepoolParams parses nodepool parameters in either JSON format or key=value pairs
 func parseNodepoolParams(params string) (map[string]string, error) {
 	if params == "" {
 		return nil, nil
 	}
+
+	// Check if the input looks like JSON (starts with { and ends with })
+	isJSON := strings.TrimSpace(params)[0] == '{' && strings.TrimSpace(params)[len(strings.TrimSpace(params))-1] == '}'
+
+	// Try to parse as JSON if it looks like JSON
+	if isJSON {
+		var poolParams spotNodePoolParams
+		decoder := json.NewDecoder(strings.NewReader(params))
+		decoder.DisallowUnknownFields()
+
+		if err := decoder.Decode(&poolParams); err != nil {
+			// If it looks like JSON but fails to parse, return a clean error message
+			if jsonErr, ok := err.(*json.UnmarshalTypeError); ok {
+				return nil, fmt.Errorf("invalid value for field '%s': %v", jsonErr.Field, jsonErr.Value)
+			}
+			return nil, fmt.Errorf("invalid nodepool configuration: %v", err)
+		}
+
+		// Successfully parsed as JSON, convert to map[string]string
+		result := make(map[string]string)
+
+		// Handle desired value which must be a whole number
+		switch v := poolParams.Desired.(type) {
+		case float64:
+			if v != float64(int(v)) {
+				return nil, fmt.Errorf("'desired' must be a whole number, got: %v", v)
+			}
+			if v < 0 {
+				return nil, fmt.Errorf("'desired' cannot be negative: %v", v)
+			}
+			result["desired"] = strconv.Itoa(int(v))
+		case string:
+			// For string values, verify it's a valid integer
+			if _, err := strconv.Atoi(v); err != nil {
+				return nil, fmt.Errorf("'desired' must be a whole number, got: %q", v)
+			}
+			result["desired"] = v
+		default:
+			return nil, fmt.Errorf("invalid type for 'desired': %T, expected number or string", v)
+		}
+
+		// Only set serverclass and bidprice if they are not empty
+		if poolParams.ServerClass != "" {
+			result["serverclass"] = poolParams.ServerClass
+		}
+
+		// Validate and format bidprice
+		if poolParams.BidPrice < 0 {
+			return nil, fmt.Errorf("'bidprice' cannot be negative: %v", poolParams.BidPrice)
+		}
+		// Format with up to 3 decimal places and clean up trailing zeros
+		bidPriceStr := fmt.Sprintf("%.3f", poolParams.BidPrice)
+		formattedBidPrice := strings.TrimRight(strings.TrimRight(bidPriceStr, "0"), ".")
+		// Ensure the formatted price is a valid number
+		if _, err := strconv.ParseFloat(formattedBidPrice, 64); err != nil {
+			return nil, fmt.Errorf("'bidprice' must be a valid number, got: %v", poolParams.BidPrice)
+		}
```

Comment:
I think this function should be responsible just for parsing, there should be a different function for validation.

Also, does it mean that key=value args which are parse below not validated?